pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -50,7 +50,7 @@ func (r *Registration) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (
 	if err != nil {
 		if nodeclaimutil.IsNodeNotFoundError(err) {
 			nodeClaim.StatusConditions().SetUnknownWithReason(v1.ConditionTypeRegistered, "NodeNotFound", "Node not registered with cluster")
-			return reconcile.Result{}, nil
+			return reconcile.Result{Requeue: true}, nil
```

Comment:
This should be caught with our re-queue on nodes here: https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/controllers/nodeclaim/lifecycle/controller.go#L136-L138

If we don't have a node when we first look at this, the node should be created, and re-trigger this reconciliation

Reply:
It makes sense. Let me check on that.